### PR TITLE
feature/EXLM-569

### PR DESCRIPTION
**Pull request by adobenitesh**
Jira ID: 569

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live/en/browse
- After: https://feature-EXLM-569--exlm--adobe-experience-league.hlx.live/en/browse

### CODE REVIEW

File: blocks/header/header.css [hunk 1]
```diff
@@ -926,12 +937,28 @@ header {
...
   box-sizing: border-box;
 }
+.header .profile-toggle[aria-expanded=true]{
+  background: var(--spectrum-gray-100);
+}
 .header .profile-menu,
 .header .product-dropdown,
 .header .product-dropdown a span {
   /* stylelint-disable-next-line */
   display: none;
```
Comment (noormd08) [inline blocks/header/header.css:947]: Why do we need /* stylelint-disable-next-line */ ?

File: blocks/header/header.js [hunk 2]
```diff
@@ -418,18 +419,16 @@ const signInDecorator = async (signInBlock) => {
...
-            <a data-id="sign-out">Sign Out</a>
           </div>
         </div>`,
       ),
     );
     const toggler = signInBlock.querySelector('.profile-toggle');
-    signInBlock.querySelector('[data-id="sign-out"]').addEventListener('click', async () => {
-      signOut();
-    });
+    if (signInBlock.querySelector('[data-id="sign-out"]')) {
```
Comment (noormd08) [inline blocks/header/header.js:427]: "sign-out" seems to be class name. Do we need to add it as data-id?

File: blocks/header/header.js [hunk 3]
```diff
@@ -524,6 +523,44 @@ const productGridDecorator = async (productGridBlock) => {
...
+    const profileMenuWrapper = document.querySelector('.profile-menu');
+    const communityHeading = document.createElement('h2');
+    communityHeading.textContent = 'Community';
+    if (profileMenuWrapper) {
+      profileMenuWrapper.innerHTML = `<h2>Learning</h2>${profileMenuBlock.innerHTML}`;
+      profileMenuWrapper.insertBefore(communityHeading, profileMenuWrapper.lastElementChild);
+    }
+
+    // const adobeIMSAccessToken = await adobeIMS?.getAccessToken().token;
+    const fcpd = await fetchCommunityProfileData('124');
```
Comment (noormd08) [inline blocks/header/header.js:548]: Please update this line

File: blocks/header/header.js [hunk 4]
```diff
@@ -524,6 +523,44 @@ const productGridDecorator = async (productGridBlock) => {
...
+
+    fcpd.data.menu.forEach((item) => {
+      const communityProfile = document.createElement('a');
+      communityProfile.href = item.url;
+      communityProfile.textContent = item.title;
+      profileMenuWrapper.insertBefore(communityProfile, profileMenuWrapper.lastElementChild);
```
Comment (noormd08) [inline blocks/header/header.js:554]: Add null checks for href and textContent

File: scripts/data-service/khoros-data-service.js [hunk 5]
```diff
@@ -1 +1,21 @@
-// Placeholder for Khoros Data Service
+import { communityProfileUrl } from '../urls.js';
+
+// eslint-disable-next-line
+export async function fetchCommunityProfileData(imsToken) {
+  try {
+    const response = await fetch(communityProfileUrl, {
+      method: 'GET',
+      headers: {
+        'x-ims-token': imsToken,
```
Comment (noormd08) [inline scripts/data-service/khoros-data-service.js:9]: Do we need this imsToken as param? as it is generic and we can directly use it here.

File: blocks/header/header.js [hunk 6]
```diff
@@ -524,6 +518,51 @@ const productGridDecorator = async (productGridBlock) => {
...
+    if (profileMenuWrapper) {
+      profileMenuWrapper.innerHTML = `<h2>Learning</h2>${profileMenuBlock.innerHTML}`;
+      profileMenuWrapper.lastElementChild.setAttribute('data-id', 'sign-out');
+      profileMenuWrapper.insertBefore(communityHeading, profileMenuWrapper.lastElementChild);
+    }
+
+    // const adobeIMSAccessToken = await adobeIMS?.getAccessToken().token;
+    const fcpd = await fetchCommunityProfileData('124'); // replace with adobeIMSAccessToken
+
+    fcpd.data.menu.forEach((item) => {
```
Comment (noormd08) [inline blocks/header/header.js:546]: Rename this variable and add the check the response object of both success and failure.

File: blocks/header/header.js [hunk 7]
```diff
@@ -524,6 +518,51 @@ const productGridDecorator = async (productGridBlock) => {
...
+    const communityHeading = document.createElement('h2');
+    communityHeading.textContent = 'Community';
+    if (profileMenuWrapper) {
+      profileMenuWrapper.innerHTML = `<h2>Learning</h2>${profileMenuBlock.innerHTML}`;
+      profileMenuWrapper.lastElementChild.setAttribute('data-id', 'sign-out');
+      profileMenuWrapper.insertBefore(communityHeading, profileMenuWrapper.lastElementChild);
+    }
+
+    // const adobeIMSAccessToken = await adobeIMS?.getAccessToken().token;
+    const fcpd = await fetchCommunityProfileData('124'); // replace with adobeIMSAccessToken
```
Comment (Sivaramvt) [inline blocks/header/header.js:544]: Could you place add a TODO wherever the change is required when we bring Khoros proxy endpoint

File: blocks/header/header.css (hunk 1 above)
Comment (adobenitesh) [reply blocks/header/header.css:946]: Done

File: blocks/header/header.js (hunk 3 above)
Comment (adobenitesh) [reply blocks/header/header.js]: Done

File: scripts/data-service/khoros-data-service.js (hunk 5 above)
Comment (adobenitesh) [reply scripts/data-service/khoros-data-service.js]: Done

File: blocks/header/header.js (hunk 6 above)
Comment (adobenitesh) [reply blocks/header/header.js]: Done

File: blocks/header/header.js (hunk 2 above)
Comment (adobenitesh) [reply blocks/header/header.js]: this is existing code

File: blocks/header/header.js (hunk 7 above)
Comment (adobenitesh) [reply blocks/header/header.js]: Done

File: blocks/header/header.js (hunk 4 above)
Comment (adobenitesh) [reply blocks/header/header.js]: Done

**Review by noormd08 — APPROVED** [review]

**Pull request merged**